### ISSUE TITLE
Cant access activity when using custom application class

### ISSUE THREAD

**Issue by alexelisenko** [fluttercommunity/flutter_downloader#52]
My project uses a plugin that requires a custom MainApplication class (audio_service).

```
<application
        android:name="io.flutter.mobile.app.MainApplication"
        android:label="app"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            <meta-data
                android:name="io.flutter.app.android.SplashScreenUntilFirstFrame"
                android:value="true" />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        <service android:name="com.ryanheise.audioservice.AudioService">
            <intent-filter>
                <action android:name="android.media.browse.MediaBrowserService" />
            </intent-filter>
        </service>

        <receiver android:name="android.support.v4.media.session.MediaButtonReceiver" >
            <intent-filter>
                <action android:name="android.intent.action.MEDIA_BUTTON" />
            </intent-filter>
        </receiver>

    </application>
```

When I navigate to the view that loads this plugin, I get the following error:

```
09:50:33.589 44 info flutter.tools E/AndroidRuntime( 2901): java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Application android.app.Activity.getApplication()' on a null object reference
```

Is there an easy way to reference the main application activity? Or perhaps add a check in `FlutterDownloaderPlugin.java` line 65:

```
public static void registerWith(PluginRegistry.Registrar registrar) {
        final FlutterDownloaderPlugin plugin = new FlutterDownloaderPlugin(registrar.context(), registrar.messenger());
        if(registrar.activity() != null){
            registrar.activity().getApplication().registerActivityLifecycleCallbacks(plugin);
        }
    }
```

I don't need the downloader plugin in the same view as the audio_service

**Comment by hnvn** [comment]
I believe this issue has been fixed in v1.2.2

**Comment by masewo** [comment]
Same issue over here.
